patricia/patricia.go

```diff
@@ -28,17 +51,33 @@ type Trie struct {
 	item   Item
 
 	children childList
+	option   *Option
 }
 
 // Public API ------------------------------------------------------------------
 
 // Trie constructor.
 func NewTrie() *Trie {
+	o := defaultOption()
+	return NewTrieWithOption(o)
+}
+
+// Trie constructor with option.
+func NewTrieWithOption(o *Option) *Trie {
 	return &Trie{
-		children: newSparseChildList(),
+		children: newSparseChildList(o.MaxChildrenPerSparseNode),
```

Comment:
@ishidawataru You should check whether any of the option fields is `0` and possibly set it to the default value instead of just directly using the value. What if someone just sets some of the option fields as in
```go
&Option{
    MaxPrefixPerNode: 10,
}
```
Then this line will use value `0` which is probably not what we want.